dandi/publish.py

```diff
@@ -0,0 +1,118 @@
+    )
+
+    lgr.info(
+        f"Getting folder metadata for {asset_type} with id {asset_id} from {girder_server_url}"
+    )
+    asset_id = set(flattened([asset_id]))
+    if len(asset_id) > 1:
+        raise ValueError(
+            f"Please point to a single type of dandiset entity. Got: {asset_type}"
+        )
+    asset_id = asset_id.pop()
+    # there might be multiple asset_ids, e.g. if multiple files were selected etc,
+    # so we will traverse all of them
+    dandiset_meta = _get_dandiset_meta(
+        asset_id, asset_type, client, authenticate, get_files=get_files
+    )
+    identifier = dandiset_meta["dandiset"]["identifier"]
+    with open(f"dandiset_{identifier}.json", "wt") as fp:
+        import json
+
+        json.dump(dandiset_meta, fp, indent=2)
+
+
+def _get_dandiset_meta(asset_id, asset_type, client, authenticate, get_files=True):
+    # First we access top level records just to sense what we are working with
+    top_entities = None
+    while True:
+        try:
+            # this one should enhance them with "fullpath"
+            top_entities = list(
+                client.traverse_asset(asset_id, asset_type, recursive=False)
+            )
+            break
+        except girder.gcl.HttpError as exc:
+            if not authenticate and girder.is_access_denied(exc):
+                lgr.warning("unauthenticated access denied, let's authenticate")
+                client.dandi_authenticate()
+                continue
+            raise
+    entity_type = list(set(e["type"] for e in top_entities))
+    if len(entity_type) > 1:
+        raise ValueError(
+            f"Please point to a single type of dandiset entity. Got: {entity_type}"
+        )
+    entity_type = entity_type[0]
+    if entity_type not in ("dandiset",):
+        raise ValueError(
+            f"Please point to a single type of dandiset entity. Got: {entity_type}"
+        )
+    # redo recursively
+    lgr.info(
+        "Traversing remote %ss (%s) recursively",
+        entity_type,
+        ", ".join(e["name"] for e in top_entities),
+    )
+    dandiset_meta = top_entities[0].get("metadata", {})
+    manifest = None
+    if get_files:
+        entities = client.traverse_asset(asset_id, asset_type, recursive=True)
+        files = (e for e in entities if e["type"] == "file")
+        files = flatten(files)
+        manifest = []
+        for file in files:
+            girder_url = (
+                f"https://girder.dandiarchive.org/api/v1/file/" f"{file['id']}/download"
+            )
+            r = requests.head(girder_url)
+            if r.status_code == "404":
+                raise NotFoundError(f"{girder_url} not found")
+            aws_url = r.headers["Location"]
+            aws_url = aws_url.split("?")[0]
+            file["dandiURL"] = f"https://api.dandiarchive.org/v1/asset/{file['id']}"
+            file["awsURL"] = aws_url
+            manifest.append(file)
+    dandiset_meta["assets"] = manifest
+    return dandiset_meta
```

Comment:
@satra - may be I misunderstood, on the call you said that you had extracted asset level metadata according to new schema, but I do not see that here...

Reply:
not according to new schema. this just cycles through an extracts the metadata currently on the server. for dandiset, the convert.py script updates it to the new schema. the same would have to be done for asset (not done yet).

i thought you were asking for something that recursively downloads all the metadata, and this would do that.

Reply:
Since dandi-cli should do its own extraction/upload of metadata (I guess we better shift that code to dandi-core as well?), where is that `convert.py` exactly?  https://github.com/search?q=org%3Adandi+convert.py&type=Repositories didn't gimme anything